syrian-contributors/auto-update.py

```diff
@@ -0,0 +1,142 @@
+import asyncio
+import aiohttp
+import json
+import uvloop
+import os
+import pytz
+import sys
+
+from datetime import datetime
+
+# Fetching 500 users now, can be 1000
+PAGES = 5
+PER_PAGE = 100
+GITHUB_TOKEN = sys.argv[1]
+
+HEADERS = {
+    "Accept": "application/vnd.github+json",
+    "Authorization": f"Bearer {GITHUB_TOKEN}",
+}
+
+
+def analyze_contributions(data):
+    syria_tz = pytz.timezone("Asia/Damascus")
+    now = datetime.now(syria_tz)
+    total_day = total_month = total_year = total_all_time = 0
+
+    for entry in data.get("contributions", []):
+        try:
+            date = datetime.strptime(entry["date"], "%Y-%m-%d")
+        except Exception:
+            continue
+        count = entry["count"]
+        total_all_time += count
+
+        if date.year == now.year:
+            total_year += count
+            if date.month == now.month:
+                total_month += count
+                if date.day == now.day:
+                    total_day += count
+
+    return total_day, total_month, total_year, total_all_time
+
+
+async def get_with_retries(session, url, headers=None, max_retries=3, delay=1):
+    for attempt in range(max_retries):
+        try:
+            async with session.get(url, headers=headers) as res:
+                if res.status == 200:
+                    return await res.json()
+                else:
+                    print(f"[!] Error {res.status} for {url}")
+        except Exception as e:
+            print(f"[!] Exception on {url}: {e}")
+        await asyncio.sleep(delay)
+    return None
+
+
+async def get_users_in_syria(session, page, semaphore):
+    url = f"https://api.github.com/search/users?q=location:Syria&type=Users&per_page={PER_PAGE}&page={page}"
+    async with semaphore:
+        data = await get_with_retries(session, url, headers=HEADERS)
+        if not data:
+            return []
+        users_only = [
+            item for item in data.get("items", []) if item.get("type") == "User"
+        ]
+        return users_only
+
+
+async def get_contributions(session, username, semaphore):
+    url = f"https://github-contributions-api.jogruber.de/v4/{username}"
+    async with semaphore:
+        data = await get_with_retries(session, url)
+        if not data:
+            return 0, 0, 0, 0
+        return analyze_contributions(data)
+
+
+async def fetch_user_data(session, username, avatar_url, semaphore):
+    print(f"🔍 Processing {username}...")
+    daily, monthly, yearly, lifetime = await get_contributions(
+        session, username, semaphore
+    )
+    return {
+        "username": username,
+        "daily_contributions": daily,
+        "monthly_contributions": monthly,
+        "yearly_contributions": yearly,
+        "total_contributions": lifetime
+        - (
+            23933 if "30838534" in avatar_url else 0
+        ),  # Execluding fake commits from user with id:30838534
```

Comment:
if you must exclude specific user id, detect via exact id or username, you can do something like the following:
```suggestion
        exclude_ids = {"30838534"}
        uid = (avatar_url.split("/")[-1] if avatar_url else "")  # brittle; better fetch user id once
```
-------

`eymeen` who is this guy? lol

Reply:
i don't know him but he have a lot of automated commits